pkg/client/unversioned/clientcmd/api/v1/types.go

```diff
@@ -54,7 +54,7 @@ type Preferences struct {
 type Cluster struct {
 	// Server is the address of the kubernetes cluster (https://hostname:port).
 	Server string `json:"server"`
-	// APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	// DEPRECATED: APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
```

Comment:
Awesome-- can you add a note here about why it's deprecated? e.g.

// Because a cluster can run multiple API groups and potentially multiple versions of each, it no longer makes sense to specify a single value for the cluster version. This field isn't really needed anyway, so we're deprecating it without a replacement. It will be ignored if it is present.